### PR TITLE
Add UI schema generator

### PR DESCRIPTION
**Pull request by edgarmueller**
Ported UI schema generator.

### CODE REVIEW

File: test/ui-schema-gen.test.ts [hunk 1]
```diff
@@ -0,0 +1,344 @@
+import test from 'ava';
+import generateDefaultUISchema, { startCase } from '../src/generators/ui-schema-gen';
+import { JsonSchema } from '../src/models/jsonSchema';
+import { Layout, ControlElement } from '../src/models/uischema';
+
+test('startCase', t => {
```
Comment (eneufeld) [inline test/ui-schema-gen.test.ts]: needed?

File: src/generators/ui-schema-gen.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    if (uiSchema['elements'] === undefined) {
+        const verticalLayout: Layout = createLayout(layoutType);
+        verticalLayout.elements.push(uiSchema);
+        return verticalLayout;
+    }
+    return <Layout>uiSchema;
+};
+
+const generateUISchema =
+    (jsonSchema: any, schemaElements: UISchemaElement[],
```
Comment (eneufeld) [inline src/generators/ui-schema-gen.ts]: should we use the JSONSchema Interface here?

File: test/ui-schema-gen.test.ts [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+        ]
+    };
+    t.deepEqual(generateDefaultUISchema(schema), uiSchema);
+});
+
+test('ignore json-schema id attributes', t => {
+    const schema: JsonSchema = {
+        type: 'object',
+        properties: {
+            id: 'ignore me',
```
Comment (edgarmueller) [inline test/ui-schema-gen.test.ts]: Schema is actually invalid, value must be an object. Remove test all together.

File: src/generators/ui-schema-gen.ts (hunk 2 above)
Comment (edgarmueller) [inline src/generators/ui-schema-gen.ts]: Yep, makes sense.

File: test/ui-schema-gen.test.ts (hunk 1 above)
Comment (edgarmueller) [inline test/ui-schema-gen.test.ts]: Ironically, the test case failures I had after porting the generator were all related to labels ;)
So, I'd argue to just leave it as is.

**Pull request merged**